### PR TITLE
Update Puppet Providers padlocal

### PR DESCRIPTION
**Pull request by Soumi7**
Updates Puppet Providers - Padlocal section in Puppet Providers

References issue #890

Screenshots :

image

image

image

image

image

image

image

image

image

image

image

image

image

### CODE REVIEW

File: docusaurus/docs/puppet-providers/padlocal.md [hunk 1]
```diff
@@ -18,57 +18,57 @@ sidebar_label: PadLocal
 
 PadLocal is a full-featured Wechaty Puppet Provider.
 
-功能 | padlocal
+Function | padlocal
 ---|---
-**<消息>**|
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/puppet-providers/padlocal.md:23]: Is there a way we can keep both Chinese and English versions 🤔 ??

But if not, I think the English version would definitely be a better choice : )

File: docusaurus/docs/puppet-providers/padlocal.md [hunk 2]
```diff
@@ -126,14 +126,64 @@ npm start
 </TabItem>
 </Tabs>
 
-## Roadmap
+## History
 
-- to be added
+In 2019 to 2020, the padlocal development team started making community-related products, and also started using Wechaty. Gradually their recognition and trust with Wechaty and the entire community was built up.
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/puppet-providers/padlocal.md:131]: Community is plural, so should we use `were` instead?

File: docusaurus/docs/puppet-providers/padlocal.md [hunk 3]
```diff
@@ -126,14 +126,64 @@ npm start
...
 
-- [New Wechaty Puppet Service: PadLocal, Padlocal, Oct 12, 2020](https://wechaty.js.org/2020/10/12/puppet-padlocal-intro/)
+### PadLocal: Wechaty puppet service provider
+
+The puppet in the community was not perfect at that time. There were some gaps in the process of using it. Moreover, the life cycle of puppet is not particularly stable, and it may not be maintained after a period of time. They were worried that this uncertainty would affect their business, so they had the idea of ​​developing a puppet themselves.
+
+So unpacking from the App, using IDA to decompile WeChat, started a long reverse road. Before doing this, the team expected this process to be difficult and encounter all kinds of difficult roadblocks, but never expected it to be so difficult. The initial stage wass quite pleasant, because we could easily see our growth every step of the way, and every time we fought a monster, we felt that we had risen to a level. But on going deeper, a lot of really hard roadblocks turned up. They even considered ​​giving up, but they were not reconciled, and persevered. Here are a few moments of those hardships:
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/puppet-providers/padlocal.md]: `wass` -> `was` ?

File: docusaurus/docs/puppet-providers/padlocal.md [hunk 4]
```diff
@@ -126,14 +126,64 @@ npm start
...
+PadLocal server will not save any business data, and there is no risk of data security.
+
+### Topology
+
+The topology diagram of the overall architecture is as follows:
+
+![Topology diagram](../../static/img/docs/puppet-services/padlocal-topological-graph.png)
+
+By implementing a puppet, the team gained a lot of things ourselves. They now have a more in-depth understanding of Wechaty, and can understand the original intention of the designer and the trade-offs among them. The fact that Wechaty is so easy to use is the result of careful design. Secondly, creating a Wechaty puppet is a very challenging thing, of course it is full of accomplishment to be able to complete such a thing. Furthermore, from an internal perspective, a more in-depth and comprehensive understanding of the WeChat terminal Operating mechanism and design ideas can be had. As a national-level communication software, WeChat's design is excellent. Various mechanisms and design concepts can fully serve as industry standards. It is worthy of being the absolute king in this field.
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/puppet-providers/padlocal.md]: of course -> ofcourse

**Review by Rohitesh-Kumar-Jain — COMMENTED** [review]
I felt that this documentation is a bit verbose and too much detailed, especially the history section. @proudofsimin what do you think ?

File: docusaurus/docs/puppet-providers/padlocal.md (hunk 1 above)
Comment (Rohitesh-Kumar-Jain) [reply docusaurus/docs/puppet-providers/padlocal.md:23]: I think the Chinese version can come in one of the Columns, that I suggest is 

|english | Chinese | tick/cross |

Comment (Soumi7) [reply docusaurus/docs/puppet-providers/padlocal.md:23]: Yes, thanks for the suggestion, will do this.

File: docusaurus/docs/puppet-providers/padlocal.md (hunk 2 above)
Comment (Soumi7) [reply docusaurus/docs/puppet-providers/padlocal.md:131]: i think community is singular, communities is plural.

File: docusaurus/docs/puppet-providers/padlocal.md (hunk 3 above)
Comment (Soumi7) [reply docusaurus/docs/puppet-providers/padlocal.md]: Changed, thanks!

File: docusaurus/docs/puppet-providers/padlocal.md (hunk 4 above)
Comment (Soumi7) [reply docusaurus/docs/puppet-providers/padlocal.md]: Changed, thanks!

**Comment by Soumi7** [comment]
Have added the english and chinese columns in the features table @Rohitesh-Kumar-Jain . It looks like this now :

image

**Comment by proudofsimin** [comment]
> I felt that this documentation is a bit verbose and too much detailed, especially the history section. @proudofsimin what do you think ?

Yes Soumi has done a bit too much for reference. The background and theory are more like within explanation, but I feel this is still ok. Thank you.

**Review by proudofsimin — APPROVED** [review]

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
LGTM, thank you for the document : )

**Review by shraddhavp — APPROVED** [review]

**Review by huan — APPROVED** [review]
LGTM.

Please read and follow #1257, and feel free to help yourself to merge it by following the merge workflow.

Thank you very much!

**Pull request merged**